maintainers/testing.md

```diff
@@ -87,6 +87,28 @@ Overall goal, follow the open source ecosystem for infrastructure choices
     - I.E. cuDF PyTests:
       - [https://github.com/rapidsai/cudf/tree/branch-0.7/python/cudf/tests](https://github.com/rapidsai/cudf/tree/branch-0.7/python/cudf/tests)
 
+## Datasets
+
+- Many tests depend on the presence of specific datasets in order to properly verify code correctness.
+- Some datasets are small enough that they can be included with the source code, but others are relatively large and cannot practically coexist with the source. In both cases, the test executable should not assume datasets are located in a particular place (either with a relative path, since the executable could be moved, or an absolute path, since different machines will likely use different absolute paths), so the location of the datasets must be provided at test runtime.
+- There are three mechanisms that allow tests to locate datasets on disk:
+  - Users can set the environment variable `RAPIDS_DATASET_ROOT_DIR` to the path of the directory containing the datasets needed by tests.
+  - Users can pass the flag `-DRAPIDS_DATASET_ROOT_DIR=<root dir>` to the `cmake` command when building from source. If the environment variable described above is not set, this setting will be used.
+  - Finally, test developers can specify a default directory to use in the event that both the environment variable and the build option are not set. Since this is essentially hardcoding a path in the test executable, users should be made aware of what the location is as well as the options they have (the `RAPIDS_DATASET_ROOT_DIR` env var and the `-DRAPIDS_DATASET_ROOT_DIR=<root dir>` build option) for overriding it.
```

Comment:
Hmmm, shouldn't there be a default value for `RAPIDS_DATASET_ROOT_DIR` so that test developers don't *have* to configure it just in case it's not set?  Convention over configuration with option to override...

Reply:
That's probably a good idea. This replaces hardcoded instances of `/datasets`, so I suppose that's a decent default given the history. Thoughts?

Another option is to throw an exception if unset, forcing the user to know about it. This might not be too bad given that the build option also exists (a one time set vs. an env var).

Reply:
The latest API uses a default value now of `/datasets`, which users can override with either the build option or the env var, or both.  https://github.com/rapidsai/cugraph/pull/215